UltimateMedals.as

```diff
@@ -328,8 +363,35 @@ void LoadFont() {
 	}
 }
 
+void UpdateText() {
+	array<string> names = {
+#if TMNEXT||MP4
+		authorText,
+#elif TURBO
+		stmasterText,
+		sgoldText,
+		ssilverText,
+		sbronzeText,
+		tmasterText,
+#endif
+		goldText,
+		silverText,
+		bronzeText,
+		pbestText
+	};	
+
+	for(uint i = 0; i < times.Length; i++) {
+		if(times[i].hidden) {
+			continue;
+		}
+
+		times[i].name = names[i];
+	}

```

Comment:
I wonder what would happen if the medals were in a different order (say PB was a silver medal). Also, I think skipping hidden seems ultimately unnecessary, so to solve the sorting issue, why not just unroll the loop:
```c
gold.name = goldText;
silver.name = silverText;
```
And so on.

Reply:
This is way smarter than my approach! It fixes sorting and skipping hidden is indeed not necessary to update the names.
```cpp
void UpdateText() {
#if TMNEXT||MP4
	author.name = authorText;
#elif TURBO
	stmaster.name = stmasterText;
	sgold.name = sgoldText;
	ssilver.name = ssilverText;
	sbronze.name = sbronzeText;
#endif
	tmaster.name = tmasterText;
	gold.name = goldText;
	silver.name = silverText;
	bronze.name = bronzeText;
	pbest.name = pbestText;
}
```